CLEBIC 2025 workshop demo notebook

aidanc151 opened the pull request:
I wanted to upload the notebook I demoed as part of @jlaehne 's HyperSpy-LumiSpy presentation at the recent CLEBIC workshop in Glasgow.

P.S. there's a few commits as I put my data files in the wrong directory

aidanc151 committed:
Add files via upload

aidanc151 committed:
Add files via upload

aidanc151 committed:
Create temp.txt

aidanc151 committed:
Delete demo-files/AlN_HR_streak_image.hspy

aidanc151 committed:
Delete demo-files/GaAsSb_streak_image.hspy

aidanc151 committed:
Delete demo-files/InGaN_hyperspectral.dm4

aidanc151 committed:
Delete demo-files/InGaN_transient_map.hspy

aidanc151 committed:
Add files via upload

aidanc151 committed:
Delete demo-files/CLEBIC2025/temp.txt

jlaehne commented:
Thanks @aidanc151 

Could you update with a version where all outputs are cleared before comitting?

PS: You can use `git rebase` to remove unneccesary commits.
See https://hyperspy.org/hyperspy-doc/current/dev_guide/git.html#keeping-the-git-history-clean

jlaehne commented:
`%matplotlib qt` needs an explaining comment, because some environments such as the linked binder cannot handle the qt interface.

`transient1.plot()` appears before it is defined

The combined figure from the last cell is not plotted in the binder notebook.

We should maybe add `numba` and `numexpr` to the `environment.yml` file to avoid warnings when running via binder.

aidanc151 committed:
Installed numba and numexpr to .yml

aidanc151 committed:
Updated notebook 

Added %matplotlib qt5 explanation
removed transient1.plot() cell
cleared all cells outputs with jupyter lab

aidanc151 committed:
Installed numba and numexpr to .yml (corrected)

aidanc151 commented:
This notebook doesn't make a lot of sense without the use of qt or widget... can binder use such backends / should I make these additional dependencies in the .yml ?

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "spectra_map.original_metadata"
```
```suggestion
    "spectra_map.metadata"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "3bf2c4b3-5b93-4431-985e-6e17c86dc4ae",
+   "metadata": {},
+   "source": [
+    "*(In the following, we will use the preprocessed dataset `transient_map` saved in the HyperSpy format. The sample contains (In,Ga)N measured by time-correlated single photon counting by Aidan Campbell at the Paul Drude Institute, Berlin.)*"
```
```suggestion
    "*(In the following, we will use the preprocessed dataset `transient_map` saved in the HyperSpy format. The sample contains an (In,Ga)N layer measured by time-correlated single photon counting by Aidan Campbell at the Paul Drude Institute, Berlin.)*"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "129fa83d-9d12-434b-b8ad-26915f4de83f",
+   "metadata": {},
+   "source": [
+    "For this dataset we'll obtain the effective lifetimes from our transients.\n",
```
```suggestion
    "For this dataset, we'll obtain the effective lifetimes from our transients.\n",
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "1016cf8c-8bd0-4621-a171-a99a03f086d7",
+   "metadata": {},
+   "source": [
+    "Models can also be created easily from custom expressions.\n",
+    "\n",
+    "New models can also be defined from custom expressions, we define one for our data as the convolution of a single exponential with a Gaussian - for the instrument response and signal decay, respectively. *Such a component will in the future be directly available in LumiSpy.*"
```
```suggestion
    "New components can also be defined from custom expressions, we define one for our data as the convolution of a single exponential with a Gaussian - for the instrument response and signal decay, respectively. *Such a component will in the future be directly available in LumiSpy.*"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "bacb1133-8472-4cdc-b9ce-4fba588e07d6",
+   "metadata": {},
+   "source": [
+    "Inspect fitting of model:"
```
```suggestion
    "To inspect the fit, we can plot our model:"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "e7d6c034-36d5-4d84-9916-be5914b16af0",
+   "metadata": {},
+   "source": [
+    "We now have a fit for all model variables for a transient at every position. We assign the values obtained for `tau` to a new signal object:"
```
```suggestion
    "We now have a fit for all model variables for a transient at every position. We assign the values obtained for the lifetime `tau` to a new signal object:"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   "id": "1aed43ab-c82f-469d-9766-90c9dfeca57e",
+   "metadata": {},
+   "source": [
+    "The streak images are of the `LumiTransientSpectrum` signaxl class:"
```
```suggestion
    "The streak images are of the `LumiTransientSpectrum` signal class:"
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+   },
+   "outputs": [],
+   "source": [
+    "# We convert our time axis to be a navigational one, \n",
+    "# therefore our signal becomes a list of spectra \n",
+    "# signals instead of on a image\n",
```
```suggestion
    "# We convert our time axis to a navigation axis, \n",
    "# thus our signal becomes a series of spectra \n",
    "# (1D signal) instead of an image (2D Signa):)\n",
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+    "# We convert our time axis to be a navigational one, \n",
+    "# therefore our signal becomes a list of spectra \n",
+    "# signals instead of on a image\n",
+    "spectral_fit = streak02.transpose(signal_axes=(0,), \n",
+    "                                  navigation_axes=(1,)).create_model()\n",
```
```suggestion
    spectral_fit = streak02.time2nav().create_model()
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+    "spectral_fit = streak02.transpose(signal_axes=(0,), \n",
+    "                                  navigation_axes=(1,)).create_model()\n",
+    "\n",
+    "g1, g2 = hs.model.components1D.Gaussian(),hs.model.components1D.Gaussian()\n",
```
```suggestion
    "g1, g2 = hs.model.components1D.Gaussian(), hs.model.components1D.Gaussian()\n",
```

jlaehne commented on Workshops/250410_CLEBIC_HyperSpy-LumiSpy.ipynb:
```diff
@@ -0,0 +1,766 @@
+    "import lumispy as lum\n",
+    "\n",
+    "# Some environments (like the linked Binder) cannot support the Qt interface, so it may not work properly in those cases.\n",
+    "# Consider using alternative backends like %matplotlib inline or %matplotlib notebook for compatibility in such environments.\n",
```
```suggestion
    "# Consider using alternative backends like %matplotlib inline or %matplotlib notebook for compatibility in such environments.\n",
    "# However, note that some of the presented interactive functionalities will not work with `inline` plotting.\n",
```

jlaehne committed:
Apply suggestions from code review

jlaehne commented:
> This notebook doesn't make a lot of sense without the use of qt or widget... can binder use such backends / should I make these additional dependencies in the .yml ?

Indeed, I amended your comment a little to this end. Binder can unfortunately only do inline plotting

jlaehne commented:
To speed up things, I committed my review comments directly, so that I can merge the PR.

jlaehne commented:
Hm, I messed up the notebook when replacing a codeline (using the convenience wrapper `time2nav()` for the streak image before fitting the spectra) and merged too fast: fixed in #23
